### PR TITLE
Fix inconsistent use of model verbose_name in permissions UI

### LINKED ISSUE #10982: Inconsistent use of model verbose_name in permissions UI
### Issue Summary

The object permissions UI doesn't show a model's [`verbose_name`](https://docs.djangoproject.com/en/4.2/ref/models/options/#verbose-name) if it differs from the model's name when it was created.

### Steps to Reproduce

Start a new project with `wagtail start myproject` and create a simple model like the following:

```py
from django.db import models

from wagtail.snippets.models import register_snippet

@register_snippet
class MyModel(models.Model):
    text = models.TextField()
```

In the group edit view, permissions render like this; note "My model" and "Can view":

image

Now, give the model a different `verbose_name`:

```py
@register_snippet
class MyModel(models.Model):
    text = models.TextField()

    class Meta:
        verbose_name = "Something else"
```

Now, the permissions render in a surprising way; note "Something else" and "Can view my model":

image

As a user, I would expect to see "Can view" the way all other objects are typically listed.

This behavior seems to come from this line that defines the permission "name":

https://github.com/wagtail/wagtail/blob/a8bc03dd8aad9d15fd4d8e84ace1bc89cbe96053/wagtail/users/templatetags/wagtailusers_tags.py#L101-L103

This code assumes that a model's content type's name (`perm.content_type.name`) will exactly match the name on the permission object (`perm.name`). This is true as long as the model's verbose name hasn't been changed, but, [per the docs](https://docs.djangoproject.com/en/4.2/ref/contrib/contenttypes/#django.contrib.contenttypes.models.ContentType.name), `content_type.name` "is taken from the `verbose_name` attribute of the model". So if a model's verbose name is changed, that change will show up when you look at its content type.

The issue is that when a model is first created, its automatically-generated permissions are given names that contain the model's  name at that time. For example, the above `MyModel` class generated permissions like this:

```
110|28|add_mymodel|Can add my model
111|28|change_mymodel|Can change my model
112|28|delete_mymodel|Can delete my model
113|28|view_mymodel|Can view my model
```

In the above example, the code assumes that since the content type reports its name as `Something else`, the permission's name will be `Can view something else`, but this isn't a valid assumption. The permission's name is still what it was when it was created, `Can view my model`. 

To fix this, the code could lookup the model objects corresponding to the permission content types and use those to do the substitution, maybe something like this:

```diff
diff --git a/wagtail/users/templatetags/wagtailusers_tags.py b/wagtail/users/templatetags/wagtailusers_tags.py
index c188425ad0..3da81fe0dd 100644
--- a/wagtail/users/templatetags/wagtailusers_tags.py
+++ b/wagtail/users/templatetags/wagtailusers_tags.py
@@ -95,11 +95,15 @@ def format_permissions(permission_bound_field):
                 }
             else:
                 extra_perms_exist["custom"] = True
+                perm_model_class = perm.content_type.model_class()
                 custom_perms.append(
                     {
                         "perm": perm,
                         "name": re.sub(
-                            f"{perm.content_type.name}$", "", perm.name, flags=re.I
+                            f"{perm_model_class._meta.model_name}$",  "", perm.name, flags=re.I
                         ).strip(),
                         "selected": checkbox.data["selected"],
                     }
```

### Technical details

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes
-   Python version: 3.11.1
-   Django version: 4.0.10
-   Wagtail version: 5.2a0

### PR DESCRIPTION
**Pull request by FatGuyy**
Fixes #10982

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [x] For Python changes: Have you added tests to cover the new/fixed behaviour? Yes, I have
    -   [x] **Firefox 118.0.1 (64-bit),  my os : Linux mint**
 -   [ ] For new features: Has the documentation been updated accordingly? Not yet.

**I think more tests might be better, but not sure how to add more tests**.

### CODE REVIEW

File: wagtail/admin/views/account.py
Comment (laymonage) [inline wagtail/admin/views/account.py:148]: This is unrelated, please rebase your branch and ensure you don't have unrelated commits.

**Review by laymonage — CHANGES_REQUESTED** [review]

File: wagtail/search/static/wagtailsearch/js/query-chooser-modal.js
Comment (lb-) [inline wagtail/search/static/wagtailsearch/js/query-chooser-modal.js:1]: This file doesn't need to be added, it's probably just because you have switched branches recently and we no longer ignore this folder. 

It's a bit confusing I know but it can be deleted.

File: wagtail/users/templatetags/wagtailusers_tags.py
Comment (lb-) [inline wagtail/users/templatetags/wagtailusers_tags.py:101]: I don't think this will solve the problem. Have another read of the issue in full. It explains that we need to try to resolve the model's verbose name here, with some discussion about different approaches. 

Hard-coding a specific label is probably not what we want to do.

**Review by lb- — CHANGES_REQUESTED** [review]
Thanks for putting this PR up with what you have done, it's going to need a bit of work still but you are in the right area of code. 

I'd recommend you read through the issue a few times and be sure you can manually reproduce, with creating some custom permissions, the problem. 

It will really help you if you can change some code, refresh, see what it's done in the bakery demo and then refine your solution. 

The unit tests will also need a bit more work to cover some of the scenarios of custom permissions with different names. 

Thanks again, hopefully you're up for continuing this.

**Comment by FatGuyy** [comment]
> Thanks for putting this PR up with what you have done, it's going to need a bit of work still but you are in the right area of code.
> 
> I'd recommend you read through the issue a few times and be sure you can manually reproduce, with creating some custom permissions, the problem.
I can reproduce it. But I'll read through it again as i am not getting a few things exactly

> It will really help you if you can change some code, refresh, see what it's done in the bakery demo and then refine your solution.
sorry, I forgot to add screenshots of the issue and solution.

> The unit tests will also need a bit more work to cover some of the scenarios of custom permissions with different names.
I am trying to but don't know how to add models in the tests.

> Thanks again, hopefully you're up for continuing this.
Thank you sir! I'm going to continue on this and whole wagtail contributions for that matter, but college submissions and intern work was too much these past few weeks

File: wagtail/admin/views/account.py
Comment (FatGuyy) [reply wagtail/admin/views/account.py:148]: Okay, I'll do that along with my next commit. Thank you sir!

Comment (FatGuyy) [reply wagtail/admin/views/account.py:151]: Hello sir, Should I create a new pr, as everything I do now stacks on top of this one

**Comment by FatGuyy** [comment]
Hello, I m trying since 2 days to understand what exactly do I need to do. It got pretty confusing. Can someone brief me about what exactly is the desired output?
@lb- @thibaudcolas @laymonage

**Comment by lb-** [comment]
Hey @FatGuyy - no need to keep pinging core team members. I and others have seen your comments but none of us have had the time to review again sorry. 

I'll put it on my radar but I may not be the best person to review either. 

I'd encourage you to sit tight (wait for a reply) or look at something else to contribute with. We talk more about this here https://docs.wagtail.org/en/stable/contributing/first_contribution_guide.html#next-steps

---


As an aside, we do encourage contributors to create a seperate branch for any PRs. This way you can easily explore other solutions to different issues without having this 'WIP' in your master branch. 

Feel free to create a new branch for this issue and raise a new PR, then close this PR. For more details see https://docs.wagtail.org/en/stable/contributing/first_contribution_guide.html#create-a-fresh-branch-for-your-contributions

File: wagtail/admin/views/account.py
Comment (lb-) [reply wagtail/admin/views/account.py:151]: As per other comments, you will either need to create a new branch that has the changes only for this issue or revert / make the changes on this branch that is only for the one issue.

File: wagtail/search/static/wagtailsearch/js/query-chooser-modal.js
Comment (lb-) [inline wagtail/search/static/wagtailsearch/js/query-chooser-modal.js:1]: This file can be deleted, it's been included accidentally.

**Review by lb- — CHANGES_REQUESTED** [review]
@FatGuyy I've just had another look and I cannot see that this PR is really ready for a review yet. It doesn't appear that you have addressed the existing comments from myself and others in reviews. 

Is there something particularly you wanted to ask about or are stuck with. We want to help you but also you'll need to take the time to read each comment and respond/ update the code. Plus have a good read of the original issue, the code suggestions on it and make sure you can accurately reproduce the problem in your local installation.

File: wagtail/search/static/wagtailsearch/js/query-chooser-modal.js
Comment (FatGuyy) [reply wagtail/search/static/wagtailsearch/js/query-chooser-modal.js:1]: Sure!
And sorry, I have been busy with my university exams.

Didn't make the PR because, I didn't have any changes to make. I'll take a read again and try to complete this PR/issue

**Pull request closed without merging**
